### DISCUSSION TITLE
Aggregation layers - Working with pre-aggregated data

### DISCUSSION THREAD

**Discussion by enderortak** [visgl/deck.gl#5900] [Q&A] [answered]
Hi. I want to visualize a huge data set, and I do not think it is a good idea to aggregate this dataset in runtime due to memory issues. Is there a way to use pre-aggregated data for aggregation layers? 

For example, to create a hexagon layer, I want to feed an array like [{centroid, numberOfPoints}].

**Comment by Pessimistress** [comment, marked as answer]
[ColumnLayer](https://deck.gl/docs/api-reference/layers/column-layer)

**Reply by enderortak** [reply]
Thank you! And is there an equivalent of heatmap?

**Reply by Pessimistress** [reply]
What kind of data are you trying to use HeatmapLayer with? Will `getPosition` and `getWeight` not achieve what you are describing?

**Reply by enderortak** [reply]
Again I want to provide an array like : [{centroid, numberOfPoints}] or [{position, weight}]. I do not want to provide the whole location data. Instead I want to provide binned data.

**Reply by Pessimistress** [reply]
Use `getPosition` and `getWeight`. If you are using SUM mode then there's no difference in the output whether your weights are 1 or pre-aggregated. If you want to use MEAN mode then you will have to use the original input. The layer generates a continuous image and you can't move that process to the CPU.

**Reply by enderortak** [reply]
Thank you! Yeah, I misunderstood the use of getPosition and getWeight functions. I thought they are callback functions which return the calculated centroids and weights from raw data. Everything is clear now :pray:
